community/values/src/main/java/org/neo4j/values/storable/CharValue.java

```diff
@@ -96,6 +97,39 @@ public int length()
         return 1;
     }
 
+    @Override
+    public TextValue substring( int start, int end )
+    {
+        assert start == 0;
+        assert end == 1;
+        return this;
+    }
+
+    @Override
+    public TextValue trim()
+    {
+        if ( Character.isWhitespace( value ) )
```

Comment:
One thing I was thinking about, is that perhaps we should make CharValue encapsulate a (32 bit) codepoint rather than a (16 bit) `char`.

(I couldn't make that comment on the relevant line of this PR, so I added it to the place that made me think of it)

Reply:
Yes let's do that as separate work